addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicBridgeHandler.java

```diff
@@ -0,0 +1,445 @@
+			thingId = entry.getValue().getThing().getUID().getId();
+			if (isConnect) {
+				logger.trace("Get Data for '{}'", thingId);
+				sendSocketData("get " + thingId);
+			}
+
+		}
+	}
+
+	// Methods for ThingHandler
+
+	public void scanThings() {
+		logger.trace("Job: Discover Things...");
+		if (!isDiscover) {
+			sendSocketData("list");
+			isDiscover = true;
+		}
+
+	}
+	
+	public ThingStatus getStatus() {
+		return getThing().getStatus();
+	}
+	
+	public void updateChannel(String thingId, String channelId, String value) {
+		sendSocketData("set "+thingId+":"+ channelId + " "+value);	
+	}
+
+	// internal Methods 
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		// No channels - nothing to do
+	}
+
+	@Override
+	public void initialize() {
+		logger.debug("Initializing Vitotronic bridge handler {}",
+				this.toString());
+		updateStatus(ThingStatus.OFFLINE);
+		VitotronicBindingConfiguration configuration = getConfigAs(VitotronicBindingConfiguration.class);
+		ipAddress = configuration.ipAddress;
+		port = configuration.port;
+		refreshInterval = configuration.refreshInterval;
+
+		isDiscover = false;
+		startAutomaticRefresh();
+	}
+	
+	@Override
+	public void dispose() {
+		logger.debug("Dispose Vitottronic bridge handler{}",
+				this.toString());
+
+		if (pollingJob != null && !pollingJob.isCancelled()) {
+			pollingJob.cancel(true);
+			pollingJob = null;
+		}
+	}
+
+	// Connection to adapter
+
+	private void openSocket() {
+		logger.trace("Try to open connection to Optolink Adapter {}:{}",
+				ipAddress, port);
+
+		try {
+			socket = new Socket(ipAddress, port);
+			out = new PrintStream(socket.getOutputStream());
+			inStream = socket.getInputStream();
+		} catch (UnknownHostException e) {
+			logger.error("Can't find Host: {}:{}", ipAddress, port);
+		} catch (IOException e) {
+			logger.debug("Error in communication to Host: {}:{}", ipAddress,
+					port);
+			logger.trace("Diagnostic: ", e);
+		}
+	}
+
+	Runnable socketReceiverRunnable = new Runnable() {
+
+		public void run() {
+			logger.trace("Start Background Thread for recieving data from adapter");
+			try {
+				XMLReader xmlReader = XMLReaderFactory.createXMLReader();
+				xmlReader.setContentHandler(new xmlHandler());
+				updateStatus(ThingStatus.ONLINE);
+				logger.trace("Start Parser for optolink adapter");
+				xmlReader.parse(new InputSource(inStream));
+
+			} catch (IOException e) {
+				logger.trace("Connection error from optolink adapter");
+			} catch (SAXException e) {
+				logger.trace("XML Parser Error");
+
+			}
+			updateStatus(ThingStatus.OFFLINE);
+			isConnect = false;
+			try {
+				if (!socket.isClosed()) socket.close();
+			} catch (Exception e) {
+			}
+			logger.trace("Connection to optolink adapter is died ... wait for restart");
+		}
+
+	};
+
+	private void startSocketReceiver() {
+
+		if (!isConnect) {
+
+			openSocket();

```

Comment:
If this fails, the thread should actually not be started at all. For a test, I have just configured the IP address of my router as an Vitotronic bridge. I get the following log:
```
2015-05-20 10:27:58 [DEBUG] [.b.v.h.VitotronicBridgeHandler:214  ] - Initializing Vitotronic bridge handler org.openhab.binding.vitotronic.handler.VitotronicBridgeHandler@5a35b3c9
2015-05-20 10:27:58 [DEBUG] [.c.thing.internal.ThingManager:140  ] - Thing status of thing vitotronic:bridge:d67f27be changed: StatusInfo [getStatus()=OFFLINE, getStatusDetail()=NONE, getDescription()=null]
2015-05-20 10:27:58 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (String) 192.168.3.1 to field 'ipAddress' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-05-20 10:27:58 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (Integer) 31113 to field 'port' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-05-20 10:27:58 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (Integer) 300 to field 'refreshInterval' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-05-20 10:27:58 [DEBUG] [.c.c.r.AbstractManagedProvider:135  ] - Updated element in ManagedThingProvider.
2015-05-20 10:27:58 [DEBUG] [.b.v.h.VitotronicBridgeHandler:250  ] - Error in communication to Host: 192.168.3.1:31113
2015-05-20 10:27:58 [DEBUG] [.c.thing.internal.ThingManager:140  ] - Thing status of thing vitotronic:bridge:d67f27be changed: StatusInfo [getStatus()=ONLINE, getStatusDetail()=NONE, getDescription()=null]
2015-05-20 10:27:58 [DEBUG] [.c.thing.internal.ThingManager:140  ] - Thing status of thing vitotronic:bridge:d67f27be changed: StatusInfo [getStatus()=OFFLINE, getStatusDetail()=NONE, getDescription()=null]
```
It should definitely not go ONLINE and then OFFLINE again!